src/Storage/Device/Local.php

```diff
@@ -96,12 +144,17 @@ public function upload($source, $path): bool
      *
      * @return string
      */
-    public function read(string $path): string
+    public function read(string $path, int $offset = 0, int $length = null): string
     {
         if(!$this->exists($path)) {
-            throw new Exception("File Not Found", 404);
+            throw new Exception("File Not Found");
+        }
+        if (\version_compare(PHP_VERSION, '8.0', '<')) {
+            if($length == null) {
+                $length = $this->getFileSize($path);
+            }
         }
-        return \file_get_contents($path);
+        return \file_get_contents($path, false, null, $offset, $length);
```

Comment:
If we drop PHP7.x support, can we use named arguments here?

Reply:
Agreed!